### PR TITLE
Add test verifying VB error ranges

### PR DESCRIPTION
**Pull request by 333fred**
Also adjusted errors that are currently out of the error range. Fixes https://github.com/dotnet/roslyn/issues/54267.

### CODE REVIEW

**Comment by 333fred** [comment]
@AlekseyTs for review.

File: src/Compilers/VisualBasic/Portable/Errors/Errors.vb [hunk 1]
```diff
@@ -1760,6 +1760,12 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         ERR_BadAbstractStaticMemberAccess = 37314
         ERR_UnimplementedSharedMember = 37315
 
+        ERR_UnmanagedCallersOnlyNotSupported = 37316
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Errors/Errors.vb:1763]: >ERR_UnmanagedCallersOnlyNotSupported [](http://example.com/codeflow?start=8&length=36)

Was this treated as a warning before?

File: src/Compilers/VisualBasic/Portable/Generated/ErrorFacts.Generated.vb [hunk 2]
```diff
@@ -175,7 +175,8 @@
                      ERRID.WRN_GeneratorFailedDuringGeneration,
                      ERRID.WRN_AnalyzerReferencesFramework,
                      ERRID.WRN_CallerArgumentExpressionAttributeSelfReferential,
-                     ERRID.WRN_CallerArgumentExpressionAttributeHasInvalidParameterName
+                     ERRID.WRN_CallerArgumentExpressionAttributeHasInvalidParameterName,
+                     ERRID.WRN_NextAvailable
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Generated/ErrorFacts.Generated.vb:179]: >ERRID.WRN_NextAvailable [](http://example.com/codeflow?start=21&length=23)

An addition of this value looks suspicious. Is it intended?

File: src/Compilers/VisualBasic/Portable/Generated/ErrorFacts.Generated.vb [hunk 3]
```diff
@@ -203,7 +204,8 @@
         Public Function IsHidden(code as ERRID) As Boolean
             Select Case code
                 Case ERRID.HDN_UnusedImportClause,
-                     ERRID.HDN_UnusedImportStatement
+                     ERRID.HDN_UnusedImportStatement,
+                     ERRID.HDN_NextAvailable
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Generated/ErrorFacts.Generated.vb:208]: >ERRID.HDN_NextAvailable [](http://example.com/codeflow?start=21&length=23)

Same here

File: src/Compilers/VisualBasic/Portable/Errors/Errors.vb [hunk 4]
```diff
@@ -2015,7 +2018,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         IDS_LangVersions = 56010
         IDS_ToolName = 56011
 
-        ERR_StdInOptionProvidedButConsoleInputIsNotRedirected = 56032
+        HDN_NextAvailable = 56012
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Errors/Errors.vb:2021]: >HDN_NextAvailable = 56012 [](http://example.com/codeflow?start=8&length=25)

It is not clear why leave previous INF/IDS values in the range of hidden diagnostics. I am not even sure if we have to maintain specific values for them.

**Comment by AlekseyTs** [comment]
Done with review pass (commit 3)

File: src/Compilers/VisualBasic/Portable/Errors/Errors.vb (hunk 1 above)
Comment (333fred) [reply src/Compilers/VisualBasic/Portable/Errors/Errors.vb:1763]: No, it was treated as an error. I created a local VB project and ran the modified test case to be sure, and they were treated as errors.

File: src/Tools/Source/CompilerGeneratorTools/Source/VisualBasicErrorFactsGenerator/Program.vb [hunk 5]
```diff
@@ -27,7 +27,9 @@ Friend Module Program
         Dim infoCodeNames As New List(Of String)
         Dim hiddenCodeNames As New List(Of String)
         For Each line In From l In File.ReadAllLines(inputPath) Select l.Trim
-            If line.StartsWith("WRN_", StringComparison.OrdinalIgnoreCase) Then
+            If (line.Contains("_NextAvailable", StringComparison.OrdinalIgnoreCase)) Then
```
Comment (AlekseyTs) [inline src/Tools/Source/CompilerGeneratorTools/Source/VisualBasicErrorFactsGenerator/Program.vb:30]: >Contains [](http://example.com/codeflow?start=21&length=8)

EndsWith?

Comment (333fred) [reply src/Tools/Source/CompilerGeneratorTools/Source/VisualBasicErrorFactsGenerator/Program.vb:30]: No. The line ends with `= SomeNumber`, not this.

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 4), assuming the CI is passing.

**Comment by 333fred** [comment]
@dotnet/roslyn-compiler for a second review.

**Comment by 333fred** [comment]
@dotnet/roslyn-compiler for a second quick reviwe.

**Comment by 333fred** [comment]
@dotnet/roslyn-compiler for a second review of this PR please.

**Review by RikkiGibson — APPROVED** [review]

**Pull request merged**